### PR TITLE
chore: adding examples that are currently hard-coded in dev docs

### PR DESCRIPTION
**Pull request by cprice404**
This commit copies over a lot of example JS code that is currently
hard-coded in the dev docs, into example files here that we can
validate in CI. Also hopefully makes it easier to make sweeping
changes in the future (such as our instanceof migration).

A subsequent dev docs PR will consume these and get rid of the
hard-coded examples.

### CODE REVIEW

File: examples/nodejs/cache/doc-example-files/working-with-files.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import * as fs from 'node:fs';
+import {CacheClient, CacheGetResponse, CacheSetResponse, Configurations, CredentialProvider} from '@gomomento/sdk';
+
```
Comment (malandis) [inline examples/nodejs/cache/doc-example-files/working-with-files.ts:3]: Are we going to commit this file? Otherwise main just bails without error.

Comment (cprice404) [reply examples/nodejs/cache/doc-example-files/working-with-files.ts:3]: ? sorry i'm not following your question

Comment (cprice404) [reply examples/nodejs/cache/doc-example-files/working-with-files.ts:3]: oh you're asking about the json file. hrm let me look

Comment (cprice404) [reply examples/nodejs/cache/doc-example-files/working-with-files.ts:3]: this example is really for the docs site, but i can add that file.

Comment (malandis) [reply examples/nodejs/cache/doc-example-files/working-with-files.ts:3]: Woops. Yes I meant the json file. As is we can syntax check but if we run the program, without the file, the run function will return fast.

**Review by malandis — APPROVED** [review]

**Pull request merged**